src/bioregistry/gh/new_prefix.py

```diff
@@ -228,40 +231,55 @@ def main(dry: bool, github: bool, force: bool):
         click.secho("No GitHub access token is available through GITHUB_TOKEN", fg="red")
         sys.exit(1)
 
-    issue_to_resource = get_new_prefix_issues()
-    if issue_to_resource:
-        click.echo(f"Found {len(issue_to_resource)} new prefix issues:")
-        for issue_number in sorted(issue_to_resource, reverse=True):
-            link = click.style(
-                f"https://github.com/biopragmatics/bioregistry/issues/{issue_number}", fg="cyan"
-            )
-            click.echo(f" - {link}")
-    else:
-        click.echo("Found no new prefix issues")
-
-    pulled_issues = github_client.get_issues_with_pr(issue_to_resource)
-    if pulled_issues:
-        click.echo(f"Found PRs covering {len(pulled_issues)} new prefix issues:")
-        for pr_number in sorted(pulled_issues, reverse=True):
-            link = click.style(
-                f"https://github.com/biopragmatics/bioregistry/pulls/{pr_number}", fg="cyan"
-            )
-            click.echo(f" - {link}")
-    else:
-        click.echo("Found no PRs covering new prefix issues")
-
-    # filter out issues that already have an associated pull request
-    issue_to_resource = {
-        issue_id: value
-        for issue_id, value in issue_to_resource.items()
-        if issue_id not in pulled_issues
-    }
-
-    if issue_to_resource:
-        click.echo(f"Adding {len(issue_to_resource)} issues after filter")
+    # If an issue number is given, process that specific issue
+    if issue is not None:
```

Comment:
since both branches of this if statement return an `issue_to_resource` dictionary, can you please refactor both branches into their own functions